api/__init__.py

```diff
@@ -31,102 +37,117 @@
+                m.set_payload(sms['text'].encode('us-ascii'))
+                m.add_header('Alphabet', 'ISO')
             except UnicodeEncodeError:
-                ucs_field = True
-                msg = sms['text'].encode('utf-16-be')
-                if msg_len > 70:
-                    parts_count = msg_len / 67 + (msg_len % 67 > 0)
-            with open(msg_file_lock, 'w') as f:
-                f.write('From: ' + auth.username() + '\n')
-                if ucs_field:
-                    f.write('Alphabet: UCS\n')
-                f.write('To: ' + mobile + '\n\n')
-                f.write(msg)
-                f.close
-                os.rename(msg_file_lock, msg_file)
-                os.chmod(msg_file, 0666)
-                app.logger.info('Message from %s to %s placed to the spooler %s' % (auth.username(), mobile, msg_file))
-                message_id = msg_file.split('/')[-1]
-            result['message_id'][mobile] = message_id
+                result['parts_count'] = text_len / 67 + (text_len % 67 > 0)
+                m.set_payload(sms['text'].encode('utf-16-be'))
+                m.add_header('Alphabet', 'UCS2')
+
+            with open(lock_file, 'w') as fp:
+                fp.write(m.as_string())
+
+            msg_file = lock_file[:-5]
+            rename(lock_file, msg_file)
+            chmod(msg_file, 0660)
+            app.logger.info('Message from %s to %s placed to the spooler as %s' % (auth.username(), mobile, msg_file))
+            result['message_id'][mobile] = basename(msg_file)
+
         else:
             app.logger.info('Forbidden to send message from %s to %s' % (auth.username(), mobile))
             result['message_id'][mobile] = 'Forbidden'
-    result['sent_text'] = sms['text']
-    result['parts_count'] = parts_count
+
     return result
 
 def access_mobile(mobile):
-    username = auth.username()
-    if app.config.has_key('MOBILE_PERMS') and app.config['MOBILE_PERMS'].has_key(username): 
-        if mobile in app.config['MOBILE_PERMS'].get(username):
-            return True
-        else:
-            return None
-    return True
+    if not app.config.has_key('MOBILE_PERMS'):
+        # Number access control disabled.
+        return True
+
+    return mobile in app.config['MOBILE_PERMS'].get(auth.username(), [])
 
 def validate_mobile(mobile):
-    if mobile.isdigit():
-        return True
-    return False
+    return re.match(r'^\+?\d+$', mobile) and True
 
 def bad_request(message):
     response = jsonify({'error': message})
     response.status_code = 400
     return response
-    
+
 @auth.get_password
 def get_password(username):
-    if username in app.config['USERS']:
-        return app.config['USERS'].get(username)
-    return None
+    return app.config['USERS'].get(username)
 
 @auth.error_handler
 def unauthorized():
     return make_response(jsonify({'error': 'Unauthorized access'}), 401)
 
-@app.route('/api/v1.0/sms', methods=['GET'])
```

Comment:
It's url (without @auth) for monitoring system (icinga/nagios/etc).  
```
# Monitoring (nginx.conf)
location /api/v1.1/sms {
    uwsgi_pass 127.0.0.1:5000;
    include uwsgi_params;
}   
```
It's  better to rename (and delete @auth.login_required):

```
@app.route('/api/v1.0/monitoring')
def monitoring():
    return jsonify({'monitoring': 'ok'})
```